### ISSUE TITLE
[Task] explore the issue of changing parameters value during the simulation

### ISSUE THREAD

**Issue by SarahAlidoost** [WUR-AI/diffWOFOST#85]
Currently, the parameters of diffwofost are static meaning they are fixed during the entire simulation period. In some applications, we might need to change this condition. 

An example is implemented in [DMC-MTL](https://github.com/Intelligent-Reliable-Autonomous-Systems/DMC-MTL). Earlier I commented on this in #13 as:

> The process of “updating parameters before each integration step” is similar to BMI functionality (they implemneted set/get_parameters instead of the typical set/get_variable see the code [here](https://github.com/Intelligent-Reliable-Autonomous-Systems/DMC-MTL/blob/b6a3a3edc6a9ed13b4487616f18c236e0b72d915/model_engine/model_engine/models/wofost/tensor_crop/tensor_leaf_dynamics.py#L231-L301)). This seems needed when using RNN to estimate the crop parameters. But I couldn't find any tests or examples showing that this works in practice. This also needs future discussion.

Another example is implemented in [HybridML-Phenology](https://github.com/WUR-AI/HybridML-Phenology). I have commented on this in #5 as:

>In this approach, three [physical (process-based) models](https://github.com/WUR-AI/HybridML-Phenology/tree/master/models/process_based) of phenology are used for setting the experiment, all different from the [PCSE phenology model](https://github.com/ajwdewit/pcse/blob/master/pcse/crop/phenology.py). In the hybrid approach, the chill response of those three models are replaced with a differentiable approximation using generalized logistic functions. Parameters of the models (chilling units) are estimated with an ML model (i.e. MLP). The framework used is torch.nn.Module.

In this context, we want to explore the challenges and answer:
- what does it mean to change the parameters during the simulation? see point (i) in [my comment](https://github.com/WUR-AI/diffWOFOST/issues/13#issuecomment-3244693410). 
- what would be the effect of changing a parameter value on gradients during the optimization?
- how to change engine to pause and resume a simulation or update the crop models internally see #21 and #25
- how can be this integrated in hybrid modelling see #11

**Comment by ronvree** [comment]
One thing I can imagine happening is that you would like to update your beliefs of either the model state or parameters based on new observations you obtain during a simulation. PCSE-Gym, for example, is basically an interface between WOFOST/PCSE and a reinforcement learning setting, where in each day in a simulation you would like to reconsider your situation and update your policy/management actions accordingly. The ability to "pause" or adjust (and copy) a simulation mid-way is essential here. Would including this functionality in the engine not also imply the possibility of changing parameters mid-simulation? Would it significantly complicate the implementation?
